src/config/http.rs

```diff
@@ -0,0 +1,239 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::io::Write;
+use std::net::SocketAddr;
+use std::panic;
+use std::str::FromStr;
+
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::routing::{get, post};
+use axum::{Extension, Json, Router, Server};
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc::{channel, Sender};
+
+use crate::config::MessageHubOptions;
+
+#[derive(Deserialize, Serialize)]
+struct ConfigResponse {
+    result: String,
+    message: Option<String>,
+}
+
+#[derive(Clone)]
+struct ConfigServerExtension {
+    store_directory: String,
+    configuration_ready_channel: Sender<()>,
+}
+
+pub struct HttpConfigProvider {
+    shutdown_channel: Sender<()>,
+}
+
+#[derive(Deserialize)]
+struct ConfigPayload {
+    realm: String,
+    device_id: String,
+    credentials_secret: Option<String>,
+    pairing_url: String,
+    pairing_token: Option<String>,
+}
+
+impl Default for ConfigResponse {
+    fn default() -> Self {
+        ConfigResponse {
+            result: "OK".to_string(),
+            message: None,
+        }
+    }
+}
+
+impl HttpConfigProvider {
+    pub(self) async fn config(
```

Comment:
Could be renamed to something like `write_config_file`, `set_config` or `store_config` to be more meaningful.

Reply:
`set_config` was a better fit as it's the name of the homologue function in `protobuf.rs`.